codex-ui/src/vue/menu/ContextMenu.vue

```diff
@@ -0,0 +1,93 @@
+<template>
+  <div :class="$style['context-menu']">
+    <div
+      v-if="showSearch"
+      :class="$style['context-menu-search']"
+    >
+      <Input :class="$style['context-menu-input']" />
+      <div :class="$style['context-menu-wrapper']">
+        <div :class="$style['context-menu-delimiter']"></div>
+      </div>
+    </div>
+    <div :class="$style['context-menu-scrollable']">
+      <div
+        v-for="(item, index) in items"
+        :key="index"
+      >
+        <div :class="$style['context-menu-wrapper']">
+          {{ item.name }}
+        </div>
+      </div>
+    </div>
+    <div
+      v-if="hasDelimiter"
+      :class="$style['context-menu-delimiter']"
+    ></div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import Input from '../input/Input.vue';
+
+interface Item {
+  /**
+   * Item id
+   */
+  id: number;
+
+  /**
+   * Item name
+   */
+  name: string;
+}
+
+withDefaults(
+  defineProps<{
+    /**
+     * If true, displays the input field for the search
+     */
+    showSearch?: boolean;
+
+    /**
+     * Array of items for context menu
+     */
+    items: Item[];
+
+    /**
+     * True if context menu should contain delimiter
+     */
+    hasDelimiter?: boolean;
```

Comment:
delimiter (let's call it "separator") is just an item with `type=separator`


to allow use it like this:

```ts
const items = [
  {
    type: 'default', // could be omitted,
    title: 'Item 1',
    onActivate: function(){},
    //...
  },
  {
    type: 'separator',
  },
  {
    type: 'default' // could be omitted,
    title: 'Item 1',
    onActivate: function(){},
    //...
  },
]
```

also, item should have `onActivate` callback to make it useful

Reply:
still actual